integrations/oci/oci-secrets-config-source/src/main/java/io/helidon/integrations/oci/secrets/configsource/OciSecretsConfigSourceProvider.java

```diff
@@ -0,0 +1,526 @@
+import io.helidon.config.spi.PollableSource;
+import io.helidon.config.spi.PollingStrategy;
+
+import com.oracle.bmc.auth.BasicAuthenticationDetailsProvider;
+import com.oracle.bmc.secrets.Secrets;
+import com.oracle.bmc.secrets.SecretsClient;
+import com.oracle.bmc.secrets.model.Base64SecretBundleContentDetails;
+import com.oracle.bmc.secrets.requests.GetSecretBundleRequest;
+import com.oracle.bmc.vault.Vaults;
+import com.oracle.bmc.vault.VaultsClient;
+import com.oracle.bmc.vault.model.SecretSummary;
+import com.oracle.bmc.vault.requests.ListSecretsRequest;
+
+import static io.helidon.integrations.oci.sdk.runtime.OciExtension.ociAuthenticationProvider;
+import static java.lang.System.Logger.Level.WARNING;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.time.Instant.now;
+import static java.util.concurrent.Executors.newVirtualThreadPerTaskExecutor;
+
+/**
+ * A {@link ConfigSourceProvider} that {@linkplain #create(String, Config) creates} {@link ConfigSource} implementations
+ * that interact with the Oracle Cloud Infrastructure (OCI) <a
+ * href="https://docs.oracle.com/en-us/iaas/tools/java/latest/com/oracle/bmc/secrets/package-summary.html">Secrets
+ * Retrieval</a> and <a
+ * href="https://docs.oracle.com/en-us/iaas/tools/java/latest/com/oracle/bmc/vault/package-summary.html">Vault</a> APIs.
+ *
+ * <p>To use, ensure the packaging artifact (e.g. {@code .jar} file or similar) containing this class is present on your
+ * class or module path as appropriate, and configure a meta-configuration source with a {@code type} of {@code
+ * oci-secrets}, following the usual Helidon meta-configuration rules.</p>
+ *
+ * <p>More specifically:</p>
+ *
+ * <ol>
+ *
+ * <li>Ensure you have an authentication mechanism set up to connect to OCI (e.g. a valid <a
+ * href="https://docs.oracle.com/en-us/iaas/Content/API/Concepts/sdkconfig.htm">OCI configuration file</a>)</li>
+ *
+ * <li>Ensure there is a classpath resource present named {@code meta-config.yaml}.</li>
+ *
+ * <li>Ensure the {@code meta-config.yaml} classpath resource contains a {@code source} with a {@code type} of {@code
+ * oci-secrets} that looks similar to the following, substituting values as appropriate:<blockquote><pre>sources:
+ *  - type: 'oci-secrets'
+ *    properties:
+ *      compartment-ocid: 'your vault compartment OCID goes here'
+ *      vault-ocid: 'your vault OCID goes here'</pre></blockquote></li>
+ *
+ * </ol>
+ *
+ * <p>Refer to Helidon's documentation concerning meta-configuration for more details.</p>
+ *
+ * @see ConfigSourceProvider
+ */
+public final class OciSecretsConfigSourceProvider implements ConfigSourceProvider, Weighted {
+
+
+    /*
+     * Static fields.
+     */
+
+
+    private static final Logger LOGGER = System.getLogger(OciSecretsConfigSourceProvider.class.getName());
+
+    private static final Set<String> SUPPORTED_TYPES = Set.of("oci-secrets");
+
+    private static final double WEIGHT = 300D;
```

Comment:
what is the significance for choosing 300D?  A short inline comment would be nice here.

Reply:
I followed examples of other `ConfigSourceProvider`s. To be frank I'm not sure if implementing `Weighted` is even needed. I am happy to follow consensus opinion. Should I remove it?

Reply:
Moved to `@Weight(300D)`; documented Tomas' rationale in a comment